app/Party.php

```diff
@@ -381,38 +381,173 @@ public function findLatest($limit = 10)
 
     public function attendees()
     {
-        //Tested
         return DB::select(DB::raw('SELECT SUM(pax) AS pax FROM '.$this->table));
     }
 
-    /**
-     * Laravel specific code.
-     */
-    public function scopeUpcomingEvents($query)
-    {
-        // We want to show approved events (wordpress_post_id IS NOT NULL), or where we are a host (e.g. because
-        // we created it, or if we are a network coordinator/admin.
-        $ret = $this->join('groups', 'groups.idgroups', '=', 'events.group')
-            ->join('users_groups', 'users_groups.group', '=', 'groups.idgroups')
-            ->leftJoin('events_users', function ($join) {
-                $join->on('events_users.event', '=', 'events.idevents');
-                $join->where('events_users.user', '=', Auth::user()->id);
-            })
-            ->leftJoin('users', 'users.id', '=', 'events_users.user')
-            ->where(function ($query) {
-                $query
-                    ->whereNotNull('events.wordpress_post_id')
-                    ->orWhere('events_users.role', '=', Role::HOST)
-                    ->orWhere('users_groups.role', '=', Role::HOST)
-                    ->orWhereIn('users.role', [Role::NETWORK_COORDINATOR, Role::ADMINISTRATOR]);
-            })
-            ->whereDate('event_date', '>=', date('Y-m-d'))
-            ->select('events.*')
-            ->groupBy('idevents')
-            ->orderBy('event_date', 'ASC');
+    // Scopes.  Each scope should build on a previous scope, getting more specific as we go down this file.  That
+    // isolates query logic more clearly.
+    private function defaultUserIds(&$userids) {
+        if (!$userids) {
+            $userids = [ Auth::user()->id ];
+        }
+    }
+
+    public function scopeUndeleted($query) {
```

Comment:
I thought Eloquent scoped to undeleted by default?  

That said, it can't hurt for us to make it explicit.  But just thinking that even without this scope, we would still be returned only undeleted, unless we explicitly asked for 'withTrashed()'.

Reply:
Hmmm.  I've definitely fixed some bugs where deleted records were being returned.  But it's possible that this was a soft-deleted row in a join table, which I think is not excluded by default.  Because of that kind of thing I've made it explicit.